hack/generate/gen-test-framework.sh

```diff
@@ -8,23 +8,22 @@ set -ex
 
 source hack/lib/test_lib.sh
 
-# Define vars
-DOCKERFILE="build/Dockerfile"
-
 # Go inside of the mock data test project
 cd test/test-framework
 
-# Create the required files in the mock test/test-framework just to allow run from it the oprator-sdk commands
-echo > $DOCKERFILE
-go mod init test-framework
-
-# Remove files that are required and created just to exec the commands.
-trap_add 'rm -rf $DOCKERFILE go.mod go.sum' EXIT
+# Ensure test-framework is up-to-date with current Go project dependencies.
+# NOTE: the SDK dependency version still needs updating on a new release.
+sdk_version="$(go list -m -f '{{.Version}}' github.com/operator-framework/operator-sdk)"
+../../build/operator-sdk print-deps > go.mod
+sed -i 's|github.com/operator-framework/operator-sdk\s*master||g' go.mod
+echo -e "\nreplace github.com/operator-framework/operator-sdk => ../../" >> go.mod
+go mod edit -require "github.com/operator-framework/operator-sdk@${sdk_version}"
+go build ./...
+go mod tidy
 
 # Run gen commands
 ../../build/operator-sdk generate k8s
-../../build/operator-sdk generate openapi
-
 # TODO(camilamacedo86): remove this when the openapi gen be set to false and it no longer is generated
 # The following file is gen by openapi but it has not been committed in order to allow we clone and call the test locally in any path.
-trap_add 'rm pkg/apis/cache/v1alpha1/zz_generated.openapi.go' EXIT
\ No newline at end of file
+trap_add 'rm -f pkg/apis/cache/v1alpha1/zz_generated.openapi.go' EXIT
```

Comment:
In @camilamacedo86's previous PR, we intentionally left off the `-f` flag to `rm` so that if we accidentally forget about removing these lines when we get rid of the openapi.go code generator, this will fail, and we will be reminded to remove this.